### PR TITLE
Inbox nonces and set bounty

### PR DESCRIPTION
**Pull request by sarvesh-ost**
Following items are implemented. 
1. The organization can change bounty. 
2. For inbox messages, the bounty validation should be done from inbox active process. 

Note : Timelock period for change bounty is not implemented.

### CODE REVIEW

File: contracts/gateway/GatewayBase.sol [hunk 1]
```diff
@@ -389,13 +389,14 @@ contract GatewayBase {
     }
 
     /**
-     * @notice Private function to get the nonce for the given account address
+     * @notice Internal function to get the outbox nonce for the given account
+     address
```
Comment (schemar) [inline contracts/gateway/GatewayBase.sol]: Missing `*` at start of doc line.

File: contracts/gateway/GatewayBase.sol [hunk 2]
```diff
@@ -413,6 +414,33 @@ contract GatewayBase {
...
+        bytes32 previousProcessMessageHash = inboxActiveProcess[_account];
+
+        if (previousProcessMessageHash == bytes32(0)) {
+            return 1;
+        }
+
+        MessageBus.Message storage message =
+        messages[previousProcessMessageHash];
+
+        return message.nonce.add(1);
```
Comment (schemar) [inline contracts/gateway/GatewayBase.sol]: lines 407-414 and 433-440 could possibly be extracted into a separate function.

File: contracts/gateway/GatewayBase.sol [hunk 3]
```diff
@@ -502,6 +530,12 @@ contract GatewayBase {
...
 
+    /**
+     * @notice Method allows organization to propose new bounty amount.
+     *
+     * @param _proposedBounty proposed bounty amount.
+     *
+     */
     function initiateBountyAmountChange(uint256 _proposedBounty)
         onlyOrganisation()
         external
```
Comment (schemar) [inline contracts/gateway/GatewayBase.sol]: It seems the order of external/internal/private methods is mixed up.

File: contracts/gateway/GatewayBase.sol [hunk 4]
```diff
@@ -502,6 +530,12 @@ contract GatewayBase {
         inboxActiveProcess[_account] = _messageHash;
     }
 
+    /**
+     * @notice Method allows organization to propose new bounty amount.
+     *
+     * @param _proposedBounty proposed bounty amount.
+     *
```
Comment (schemar) [inline contracts/gateway/GatewayBase.sol]: unnecessary line

File: contracts/gateway/GatewayBase.sol [hunk 5]
```diff
@@ -511,18 +545,29 @@ contract GatewayBase {
         emit BountyChangeInitiated(bounty, _proposedBounty);
     }
 
+    /**
+     * @notice Method allows organization to confirm proposed bounty amount.
+     *
```
Comment (schemar) [inline contracts/gateway/GatewayBase.sol]: unnecessary line

File: contracts/gateway/GatewayBase.sol [hunk 6]
```diff
@@ -511,18 +545,29 @@ contract GatewayBase {
...
         onlyOrganisation()
         external
+        returns(
+            uint256 changedBountyAmount_,
+            uint256 previousBountyAmount_
+        )
     {
         require(proposedBounty != bounty,
-        "Proposed bounty should be different from existing bounty");
+            "Proposed bounty should be different from existing bounty");
```
Comment (schemar) [inline contracts/gateway/GatewayBase.sol]: This should probably look like the following instead:
```solidity
require(
    proposedBounty != bounty,
    "Proposed bounty should be different from existing bounty"
);
```

File: contracts/gateway/GatewayBase.sol [hunk 7]
```diff
@@ -99,6 +97,14 @@ contract GatewayBase {
 
     /** Gateway link message hash. */
     bytes32 public gatewayLinkHash;
+    
+    /** amount of ERC20 which is staked by facilitator. */
+    uint256 public bounty;
+
+    /** Proposed new bounty amount for bounty change. */
+    uint256 public proposedBounty;
+    /** bounty proposal block height*/
```
Comment (deepesh-kn) [inline contracts/gateway/GatewayBase.sol]: new line

**Pull request merged**